File: backend-go/internal/model/converter/session.go
package converter

import (
	"kido1611/notes-backend-go/internal/db/sqlc"
	"kido1611/notes-backend-go/internal/model"
)

func SessionToResponse(session *sqlc.Session) *model.SessionResponse {
	userId := ""
	ipAddress := ""
	userAgent := ""

	if session.UserID.Valid {
		userId = session.UserID.String
	}

	if session.IpAddress.Valid {
		ipAddress = session.IpAddress.String
	}

	if session.UserAgent.Valid {
		ipAddress = session.UserAgent.String
	}

	return &model.SessionResponse{
		ID:        session.ID,
		UserID:    userId,
		CsrfToken: session.CsrfToken,
		IpAddress: ipAddress,
		UserAgent: userAgent,
		ExpiredAt: session.ExpiredAt,
	}
}

File: backend-rust/src/application/models/session.rs
use chrono::NaiveDateTime;
use serde::Serialize;
use uuid::Uuid;

use crate::domain::models::session::Session;

#[derive(Serialize, Debug)]
pub struct SessionResponse {
    pub id: String,
    pub user_id: Option<Uuid>,
    pub csrf_token: String,
    pub ip_address: Option<String>,
    pub user_agent: Option<String>,
    pub expired_at: NaiveDateTime,
}

impl SessionResponse {
    pub fn from_session_entity(session: &Session) -> Self {
        SessionResponse {
            id: session.id.clone(),
            user_id: session.user_id,
            csrf_token: session.csrf_token.clone(),
            ip_address: session.ip_address.clone(),
            user_agent: session.user_agent.clone(),
            expired_at: session.expired_at,
        }
    }
}
